pkg/graph/constructor.go

```diff
@@ -17,14 +17,154 @@ limitations under the License.
 package graph
 
 import (
+	"context"
 	"fmt"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
 	eventingv1beta3 "knative.dev/eventing/pkg/apis/eventing/v1beta3"
 	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
+func ConstructGraph(ctx context.Context, filterFunc func(obj interface{}) bool) (*Graph, error) {
+	eventingClient := eventingclient.Get(ctx)
```

Comment:
I'd get this as an input instead of assuming injection context

Reply:
the future proof way would be to have a struct as input so that we can evolve this function